nbs/examples/train_imagenette.py

```diff
@@ -13,23 +13,40 @@
 def pr(s):
     if rank_distrib()==0: print(s)
 
-def get_dls(size, woof, bs, sh=0., workers=None):
+def get_dls(size, woof, pct_noise, bs, sh=0., workers=None):
+    assert pct_noise in [0,5,50], '`pct_noise` must be 0,5 or 50.'
     if size<=224: path = URLs.IMAGEWOOF_320 if woof else URLs.IMAGENETTE_320
     else        : path = URLs.IMAGEWOOF     if woof else URLs.IMAGENETTE
     source = untar_data(path)
     workers = ifnone(workers,min(8,num_cpus()))
+    blocks=(ImageBlock, CategoryBlock)
+    tfms = [RandomResizedCrop(size, min_scale=0.35), FlipItem(0.5)]
     batch_tfms = [Normalize.from_stats(*imagenet_stats)]
     if sh: batch_tfms.append(RandomErasing(p=0.3, max_count=3, sh=sh))
-    dblock = DataBlock(blocks=(ImageBlock, CategoryBlock),
-                       splitter=GrandparentSplitter(valid_name='val'),
-                       get_items=get_image_files, get_y=parent_label,
-                       item_tfms=[RandomResizedCrop(size, min_scale=0.35), FlipItem(0.5)],
-                       batch_tfms=batch_tfms)
-    return dblock.dataloaders(source, path=source, bs=bs, num_workers=workers)
+    
+    if pct_noise > 0:
+        csv_file = 'noisy_imagewoof.csv' if woof else 'noisy_imagenette.csv'
+        inp = pd.read_csv(source/csv_file)
+        dblock = DataBlock(blocks=blocks,
```

Comment:
Could this be simplified by having a single `DataBlock` call, always using the CSV, and just using different columns for each noise level (including zero noise)?

Reply:
fixed